contracts/TxRelay.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity 0.4.8;
+
+//This contract is meant as a "singleton" forwarding contract.
+//Eventually, it will be able to forward any transaction to
+//Any contract that is built to accept it.
+
+contract TxRelay {
+
+  // Note: This is a local nonce.
+  // Different from the nonce defined w/in protocol.
+  mapping(address => uint) nonce;
+
+  /*
+   * @dev Relays meta transactions
+   * @param sigV, sigR, sigS ECDSA signature on some data to be forwarded
+   * @param data The bytes necessary to call the function in the destination contract.
+                 Note, the first encoded argument in data must be address of the signer
+   * @param claimedSender Address of the user who is having tx forwarded
+   */
+  function relayMetaTx(uint8 sigV, bytes32 sigR, bytes32 sigS,
+                       address destination, bytes data) {
+
+    address claimedSender = getAddress(data);
+    // relay :: nonce :: destination :: data :: relayer
+    bytes32 h = sha3(this, nonce[claimedSender], destination, data, msg.sender);
+    address addressFromSig = ecrecover(h, sigV, sigR, sigS);
+
+    nonce[claimedSender]++;
+
+    if (claimedSender != addressFromSig) throw;
+
+    if (!destination.call(data)) {
+      //In the future, add event here. Has semi-complex gas considerations. See EIP 150
+    }
+  }
+
+  /*
+   * @dev Gets an address encoded as the first argument in transaction data
+   * @param b The byte array that should have an address as first argument
+   * @returns a The address retrieved from the array
+     (Optimization based on work by tjade273)
+   */
+  function getAddress(bytes b) constant returns (address a) {
+    if (b.length < 36) return address(0);
+    assembly {
+        let mask := 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF
+        a := and(mask, mload(add(b,36)))
```

Comment:
Actually, cannot do this. Solidly does some weird things w/ inline assembly that causes compilation to fail. Instead, I will add comments to clarify.